test/extensions/http/header_validators/envoy_default/header_validator_test.cc

```diff
@@ -16,80 +9,281 @@ namespace HeaderValidators {
-using ::Envoy::Http::HeaderValidatorFactory;
+using ::Envoy::Http::Protocol;
+using ::Envoy::Http::RequestHeaderMap;
+using ::Envoy::Http::ResponseHeaderMap;
+
+class BaseHttpHeaderValidator : public HeaderValidator {
+public:
+  BaseHttpHeaderValidator(
+      const envoy::extensions::http::header_validators::envoy_default::v3::HeaderValidatorConfig&
+          config,
+      Protocol protocol, StreamInfo::StreamInfo& stream_info)
+      : HeaderValidator(config, protocol, stream_info) {}
 
-namespace {
+  HeaderEntryValidationResult validateRequestHeaderEntry(const HeaderString&,
+                                                         const HeaderString&) override {
+    return HeaderEntryValidationResult::success();
+  }
 
-constexpr absl::string_view empty_config = R"EOF(
-    name: envoy.http.header_validators.envoy_default
-    typed_config:
-        "@type": type.googleapis.com/envoy.extensions.http.header_validators.envoy_default.v3.HeaderValidatorConfig
-)EOF";
+  HeaderEntryValidationResult validateResponseHeaderEntry(const HeaderString&,
+                                                          const HeaderString&) override {
+    return HeaderEntryValidationResult::success();
+  }
 
-} // namespace
+  RequestHeaderMapValidationResult validateRequestHeaderMap(RequestHeaderMap&) override {
+    return RequestHeaderMapValidationResult::success();
+  }
 
-class HeaderValidatorTest : public testing::Test {
+  ResponseHeaderMapValidationResult validateResponseHeaderMap(ResponseHeaderMap&) override {
+    return ResponseHeaderMapValidationResult::success();
+  }
+};
+
+using BaseHttpHeaderValidatorPtr = std::unique_ptr<BaseHttpHeaderValidator>;
+
+class BaseHeaderValidatorTest : public HeaderValidatorTest {
 protected:
-  ::Envoy::Http::HeaderValidatorPtr create(absl::string_view config_yaml,
-                                           ::Envoy::Http::Protocol protocol) {
-    auto* factory =
-        Registry::FactoryRegistry<Envoy::Http::HeaderValidatorFactoryConfig>::getFactory(
-            "envoy.http.header_validators.envoy_default");
-    ASSERT(factory != nullptr);
-
-    envoy::config::core::v3::TypedExtensionConfig typed_config;
+  BaseHttpHeaderValidatorPtr createBase(absl::string_view config_yaml) {
+    envoy::extensions::http::header_validators::envoy_default::v3::HeaderValidatorConfig
+        typed_config;
     TestUtility::loadFromYaml(std::string(config_yaml), typed_config);
 
-    uhv_factory_ = factory->createFromProto(typed_config.typed_config(), context_);
-    return uhv_factory_->create(protocol, stream_info_);
+    return std::make_unique<BaseHttpHeaderValidator>(typed_config, Protocol::Http11, stream_info_);
   }
+};
 
-  void setHeaderStringUnvalidated(HeaderString& header_string, absl::string_view value) {
-    header_string.setCopyUnvalidatedForTestOnly(value);
-  }
+TEST_F(BaseHeaderValidatorTest, ValidateMethodPermissive) {
+  HeaderString valid{"GET"};
+  HeaderString custom{"CUSTOM-METHOD"};
```

Comment:
Please make some character's lowercase, since for customer methods it does not have to be uppercase.